GeometryTools/Size.swift

```diff
@@ -25,9 +25,20 @@ public struct Size {
         self.height = height
     }
 
+    /// - Returns: a `Size` with both dimensions scaled by the same value.
     public func scaled(by value: Double) -> Size {
         return Size(width: width * value, height: height * value)
     }
+
+    /// - Returns: a `Size` scaled by the given dimensions.
+    public func scaled(byWidth widthScale: Double = 1, byHeight heightScale: Double = 1) -> Size {
```

Comment:
Perhaps the signature could be:

```Swift
func scaledBy(width: Double, height: Double) { ... }
```

This would be analogous to the:

```Swift
point.translatedBy(x:y:)
```